processor/src/main/java/com/linecorp/decaton/processor/runtime/TracingProvider.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2020 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.decaton.processor.runtime;
+
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+
+import com.linecorp.decaton.client.KafkaProducerSupplier;
+import com.linecorp.decaton.processor.DecatonProcessor;
+import com.linecorp.decaton.processor.DeferredCompletion;
+import com.linecorp.decaton.processor.ProcessingContext;
+
+/**
+ * Interface for distributed tracing implementations that track traces via Kafka {@link ConsumerRecord}s
+ * (typically through Kafka headers) e.g. Zipkin, OpenTracing, Kamon.
+ * Note that this will usually need to be paired with a suitable {@link KafkaProducerSupplier} on the producer
+ * side to write those Kafka headers.
+ * In particular, if you want retries to be shown as part of the same (distributed) trace, make sure to supply
+ * such a {@link KafkaProducerSupplier} in your {@link RetryConfig}.
+ */
+@FunctionalInterface
```

Comment:
AFAIU, putting` @FunctionInterface` indicates that we intend to keep this interface to have just one required function to implement, which is not a case for this I think.

Reply:
I'd expect this interface will always consist of just `traceFor` and everything else will be on the `TraceHandle`.

Reply:
yeah, might be. but people can still use lambda for this class regardless to whether its annotated by `@FunctionalInterface` or not, making me to feel there's no benefit to put it except for the limited cases that we strongly intend it to be an argument for functional-style interfaces.
it's just a way of my thinking so feel free to leave it if you like. (we'd just remove it when we need to add an another method in this interface)

Reply:
Ah, I hadn't realised that. To be honest I was mainly going by IntelliJ's warning. Will remove.